opendatahub-io/ai-edge
# RHOAIENG-201: Add e2e-tests to verify the MLOps pipelines are working correctly

## Description
Adds two make targets `go-test` `go-test-setup`. Running `make go-test` with its required environment variables defined will begin the tests. A working namespace `test-namespace` is required to for both the setup and test target

## How Has This Been Tested?
- Log into a cluster with `oc`
- Set the list of enviroment variables that are needed for running the tests for both the `go-test` and `go-test-setup` make targets. All of the required fields are listed [here](https://github.com/jackdelahunt/ai-edge/blob/pipeline-testing/Makefile#L37-L44) and [here](https://github.com/jackdelahunt/ai-edge/blob/pipeline-testing/Makefile#L74-L76)
- Run `oc new-project test-namespace`
- Run `make AWS_SECRET=... AWS_ACCESS=...  ... go-test-setup`
- Run `make S3_BUCKET=...  ... go-test`
- All tests should pass

### Note
- Setting `GO` to the path to any go compiler you may have when running `go-test` will overwrite the compiler used.

## Merge criteria:
